cabal-testsuite/src/Test/Cabal/Prelude.hs

```diff
@@ -255,13 +256,16 @@ cabal :: String -> [String] -> TestM ()
 cabal cmd args = void (cabal' cmd args)
 
 cabal' :: String -> [String] -> TestM Result
-cabal' = cabalG' []
+cabal' cmd args = cabalG' [] cmd args Nothing
+
+cabalWithStdin :: String -> [String] -> String -> TestM Result
+cabalWithStdin cmd args input = cabalG' [] cmd args (Just input)
 
 cabalG :: [String] -> String -> [String] -> TestM ()
-cabalG global_args cmd args = void (cabalG' global_args cmd args)
+cabalG global_args cmd args = void (cabalG' global_args cmd args Nothing)
 
-cabalG' :: [String] -> String -> [String] -> TestM Result
-cabalG' global_args cmd args = do
+cabalG' :: [String] -> String -> [String] -> Maybe String -> TestM Result
```

Comment:
This breaks informal naming rule, the primed version returns `Result`, but otherwise is the same.

Rather make a newly named function (and thus avoid editing unrelated test cases).